x-pack/plugins/security_solution/server/lib/detection_engine/schemas/rule_converters.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  CreateTypeSpecific,
+  FullResponseSchema,
+  ResponseTypeSpecific,
+} from '../../../../common/detection_engine/schemas/request';
+import { RuleActions } from '../rule_actions/types';
+
+// These functions provide conversions from the request API schema to the internal rule schema and from the internal rule schema
+// to the response API schema. This provides static type-check assurances that the internal schema is in sync with the API schema for
+// required and defaultable fields. However, it is still possible to add an optional field to the API schema
+// without causing a type-check error here.
+
+// Converts params from the snake case API format to the internal camel case format AND applies default values where needed.
+// Notice that params.language is possibly undefined for most rule types in the API but we default it to kuery to match
+// the legacy API behavior
+export const typeSpecificSnakeToCamel = (params: CreateTypeSpecific): TypeSpecificRuleParams => {
+  switch (params.type) {
+    case 'eql': {
+      return {
+        type: params.type,
+        language: params.language,
+        index: params.index,
+        query: params.query,
+        filters: params.filters,
+        eventCategoryOverride: params.event_category_override,
+      };
+    }
+    case 'threat_match': {
+      return {
+        type: params.type,
+        language: params.language ?? 'kuery',
+        index: params.index,
+        query: params.query,
+        filters: params.filters,
+        savedId: params.saved_id,
+        threatFilters: params.threat_filters,
+        threatQuery: params.threat_query,
+        threatMapping: params.threat_mapping,
+        threatLanguage: params.threat_language,
+        threatIndex: params.threat_index,
+        concurrentSearches: params.concurrent_searches,
+        itemsPerSearch: params.items_per_search,
+      };
+    }
+    case 'query': {
+      return {
+        type: params.type,
+        language: params.language ?? 'kuery',
+        index: params.index,
+        query: params.query ?? '',
+        filters: params.filters,
+        savedId: params.saved_id,
+      };
+    }
+    case 'saved_query': {
+      return {
+        type: params.type,
+        language: params.language ?? 'kuery',
+        index: params.index,
+        query: params.query,
+        filters: params.filters,
+        savedId: params.saved_id,
+      };
+    }
+    case 'threshold': {
+      return {
+        type: params.type,
+        language: params.language ?? 'kuery',
+        index: params.index,
+        query: params.query,
+        filters: params.filters,
+        savedId: params.saved_id,
+        threshold: params.threshold,
+      };
+    }
+    case 'machine_learning': {
+      return {
+        type: params.type,
+        anomalyThreshold: params.anomaly_threshold,
+        machineLearningJobId: params.machine_learning_job_id,
+      };
+    }
+    default: {
+      return assertUnreachable(params);
```

Comment:
Took out one of these statements and the exhaustive switch statement looks good and like it is working 👍